### ISSUE TITLE
Removed panels are displayed on case page

### ISSUE THREAD

**Issue by Jakob37** [Clinical-Genomics/scout#4695]
On the panel page, we can see that the v1.0 "albinism" has been removed:

![scout_removed_panel](https://github.com/Clinical-Genomics/scout/assets/7374399/770c2d45-c203-4204-aace-8128ab0bcaca)

But it is still displayed among the panels on the case page:

![scout_removed_panel_shown](https://github.com/Clinical-Genomics/scout/assets/7374399/d9edfa86-b25d-4f8b-af7a-49f1ee490a9d)

This is true generally for removed panels.

I suspect that this is a bug - that the removed panels shouldn't be displayed.

**Comment by dnil** [comment]
With the current description, I would lean towards feature request rather than bug?

Part of the reason we do not just delete old panels is that they appear on old cases, which may make little sense without them. The case had the panel, with the stated version so we see very little reason to remove it. How about you?
If it helps some users you might add a similar badge on the case page perhaps!

**Comment by Jakob37** [comment]
> With the current description, I would lean towards feature request rather than bug?
> 
> Part of the reason we do not just delete old panels is that they appear on old cases, which may make little sense without them. The case had the panel, with the stated version so we see very little reason to remove it. How about you? If it helps some users you might add a similar badge on the case page perhaps!

Aha, OK, yes that makes sense.

Our geneticists has been worrying about this frequently though, that they might accidentally use a removed panel. Adding the "Removed" badge would prevent that.

I wonder if it might be cluttered to still show them if they have many removed panel (remember our geneticists includes many dozens of panels for wgs samples). That they might want the option to hide them. But that could be a question for another day.

I could take a look at adding the badge if you think that is a good idea?

**Comment by dnil** [comment]
Sounds great - and hiding should be all good as well, as long as one is able to unhide if some confusion arises. How to handle the panel filter drop-down on the variants filter page is less obvious to me right now,m, but feel free to suggest something there!

**Comment by Jakob37** [comment]
> Sounds great - and hiding should be all good as well, as long as one is able to unhide if some confusion arises.

Yes, that sounds good! I think an unhide in same style of the one on the panel page would work great at least for the geneticists in Lund. If that would be fine elsewhere, I would be happy to try out implementing this.

<img width="487" alt="0_panel_hide" src="https://github.com/Clinical-Genomics/scout/assets/7374399/8c659726-db51-4cff-b815-364d308a195c">

> How to handle the panel filter drop-down on the variants filter page is less obvious to me right now,m, but feel free to suggest something there!

Testing it out, it actually seems to not show the deleted panels at all at the moment. This is in constrast to the drop-down on the case page which shows deleted panels (also not alphabetically sorted at the moment). Talking further to our geneticsts today I realized that this one is the most concerning for them, as there is no way to distinguish deleted panels at the moment (below "albinism" is an old deleted panel, while "Albinism" is the one to use).

Here is the dropdown on the case page - unsorted and with deleted panels:

<img width="126" alt="scout_albi" src="https://github.com/Clinical-Genomics/scout/assets/7374399/b003c1b8-0dc2-4c25-b06f-c98ba40379b8">

Here is the dropdown on the variant page - sorted and without deleted panels: 

<img width="168" alt="scout_variants_panel" src="https://github.com/Clinical-Genomics/scout/assets/7374399/9368c823-2ce9-4ba2-9093-aa9b727aba8c">

So a suggested plan:

- Adding a default-checked "Hide removed panels" to both the case page and the variant page
- Add the "deleted" badge in the list shown on the case page, again following the style of the panel page
- Alphabetically sort the case dropdown

If the plan gets your blessing I could make a draft for further input. (I'll be away next two weeks though, so might or might not be done with it promptly).

**Comment by Jakob37** [comment]
A separate but related point which I am not sure whether an issue or not, is that while I can see the removed panels using "Hide removed panels", our lead geneticsts could not. When she clicked that button for the same institute for which I could see removed panels, nothing happened for her. She tested also in my browser with her account, so seems to be a user setting. 

Is it only admin that could see the removed panels? It would be useful for her to be able to see them.

If an issue, I can make an issue on this.

**Comment by dnil** [comment]
Sounds great! Heh, maybe it is the panel use being slightly different between the sites (and "newer" to Lund) that makes this feel a bit clunky. Good that you keep at it until we converge.

I quite agree about the main issue, and the solution of adding `Removed` badges in the case page gene panel list.
For me, I think you can keep the variantS page as-is with the removed panels hidden from search. That reflects the use well, and they are easy to "restore" if they are desired in search again. Feel free with the hiding toggle on the case page as is good for you. In Solna, we very rarely have removed panels, and if so it's only a few handfuls of panels tops per case anyway so the clutter is ok.

Let's see, a couple of points on the potential other issues.
- I see good alphabetical sort of all panel dropdowns, with some clearly awaited exceptions:
1) The panels added as part of the institute settings additional "gene panels available for variants filtering", which are added to the end of the cases' clinical panels. Maybe add a separator line/in-menu heading to the dropdown if this is deemed confusing?
2) The panels marked as default for a case, which are displayed first, internally in alphabetical order, in the list on the case page. These are already color coded, and with badges, so should not be confusing?
Am I missing something?

- Not being able to see the removed panels is at least partly apparently by design; the same `shall_display_panel()` is used, so if a panel is not visible, a user needs to have write access, which they only have to panels they are `maintainer` for, unless they are admins. I see no harm in showing these on request: on the panelS page, the option indicates they can already select if they want that extra clutter! They should still not be able to restore it without "write" permissions for it naturally. But yeah, let's make a new issue to keep the conversations easier to follow! 😅

**Comment by Jakob37** [comment]
Nice that we are the same page! Yes, it indeed seems that most of the clunkyness comes from the different ways of using panels, where in Lund we seems to use many and remove many as they are updated.

> I quite agree about the main issue, and the solution of adding `Removed` badges in the case page gene panel list. For me, I think you can keep the variantS page as-is with the removed panels hidden from search. That reflects the use well, and they are easy to "restore" if they are desired in search again. Feel free with the hiding toggle on the case page as is good for you. In Solna, we very rarely have removed panels, and if so it's only a few handfuls of panels tops per case anyway so the clutter is ok.

Sounds good! Then I leave the variantS page as is, and focus on the case page.

> Let's see, a couple of points on the potential other issues.
> 
> I see good alphabetical sort of all panel dropdowns, with some clearly awaited exceptions:

Are you looking at the list shown on the case page or the dropdown on the case page? The list looks fine for us. It is the drop-down I am claiming to be unsorted. Here is a screenshot:

<img width="192" alt="unordered_list" src="https://github.com/Clinical-Genomics/scout/assets/7374399/37efb6a0-3ae1-4e0d-8af4-ea37b8c05853">

> But yeah, let's make a new issue to keep the conversations easier to follow! 😅

New issue: #4696

**Comment by dnil** [comment]
> Are you looking at the list shown on the case page or the dropdown on the case page? The list looks fine for us. It is the drop-down I am claiming to be unsorted. Here is a screenshot:

Ah, the case page "Change default gene panel" dropdown? Yes, it does look sorted on my end, but that is probably just because the case panels happen to be neatly arranged alphabetically on load in the first place.
![Screenshot 2024-07-03 at 08 44 04](https://github.com/Clinical-Genomics/scout/assets/758570/59d44447-6b59-4edc-b6cf-07b59d0a914e)
I can take a look, since I'm already thinking about it - and sorted the others without noticing this one... 😆

**Comment by Jakob37** [comment]
> I can take a look, since I'm already thinking about it - and sorted the others without noticing this one... 😆

OK sounds great 😄
